routers/repo/compare.go

```diff
@@ -434,6 +434,8 @@ func CompareDiff(ctx *context.Context) {
 	setTemplateIfExists(ctx, pullRequestTemplateKey, pullRequestTemplateCandidates)
 	renderAttachmentSettings(ctx)
 
+	ctx.Data["IsIssueWriter"] = ctx.Repo.CanWrite(models.UnitTypePullRequests)
```

Comment:
We should rename it as `IsPullRequestWriter`

Reply:
Hello, your suggestion is reasonable, but maybe not necessary, the main problem is that the new issue and new Pull Request use same tmpl ``new_form.tmpl``, If change it , the tmpl will become complex. then we have think of Pull Request as issue in many place , so it's not a big problem. Thanks.

https://github.com/a1012112796/gitea/blob/61ceed7442b15b1c63d1cddb83ba706cb498337a/templates/repo/diff/compare.tmpl#L77-L79

https://github.com/a1012112796/gitea/blob/61ceed7442b15b1c63d1cddb83ba706cb498337a/templates/repo/issue/new.tmpl#L8-L10

Reply:
This could also be ctx.Data["HasWritePermissions"] too which I think is more clear

IsIssueWriter or IsPullRequesttWriter makes it sound like that user wrote/authored the issue or pull request when really we are just checking if they have write permissions)

Reply:
> This could also be ctx.Data["HasWritePermissions"] too which I think is more clear
> 
> IsIssueWriter or IsPullRequesttWriter makes it sound like that user wrote/authored the issue or pull request when really we are just checking if they have write permissions)

that's great but maybe not accurate enough, how about ``HasIssuesOrPullsWritePermission``?

Reply:
Sure, that would work fine too I think